### DISCUSSION TITLE
Contributor rank

### DISCUSSION THREAD

**Discussion by 0xsingular** [teaxyz/white-paper#114] [Ideas]
It would be a good perspective to create a contributor rank based on the interactions within the tea protocol (also can establish a general rank based on several criteria as well as some ranks per contribution category), for which some criteria can be:
- Donations to registered OSS projects.
- Reported vulnerabilities.
- How many TEA tokens and to how many projects are staked.
- Voting in governance proposals.

By setting a rank, users/community/contributors will be more actively involved in contributing and creating a better reputation within the protocol.
Also this rank can be used by registered projects within the protocol to set different tasks for contributors against benefits, general tasks or specific tasks that are only available to contributors with a set minimum general rank, or a set minimum category rank.

**Comment by thomas-borrel** [comment]
Thank you @0xsingular - I think we need to distinguish 2 things:
- ITN
- Mainnet
For ITN, there will be a participant scoreboard where all ITN participants can see where they are relative to the rest of the community.
Post ITN, I think some attention should continue to go towards enhancing a package level and individual level reputation system that can inform developers' choice in packages and help the entire community understand which areas each member has expertise in. I think there could be a lot of benefits there - For example, OSS project maintainers could better target their search when they need skills added to their project, or the teaDAO could form committees of experts to resolve disputes around vulnerabilities or to enhance the protocol.

**Reply by 0xsingular** [reply]
One question regarding teaDAO, does the team have a prepared model of organization and governance of teaDAO, any relevant docs (operating manual, code of conduct, constitution)? Will these be published on GitHub before the mainnet for discussion/feedback/suggestions?

**Reply by harfiansol** [reply]
lets see tea document

**Comment by 0xsingular** [comment]
I was referring to mainnet, aiming for more active community involvement and deeper understanding of what the projects are building.

**Comment by rav2525** [comment]
Good one.
I would also like to add a point here. The ranking can also include contribution of the below assets having said that there are many software developer in various domains but only end up spamming chats for ranks:
- There relevant use-case and solution based on there previous experiences, inline with current project 
- Useful assets to git repo that doesn't have sensitive data
- Improvements the project can make

**Comment by abelonuh** [comment]
looking forward to what tea will offer.

**Comment by engr0001** [comment]
great project....looking forward to what tea will offer.
